application/view/common/resource-values.phtml

```diff
@@ -32,12 +37,14 @@ $labelInfo = $this->setting('property_label_information');
                 $class[] = 'private';
             }
             ?>
-            <div class="<?php echo implode(' ', $class); ?>" lang="<?php echo $escape($value->lang()); ?>">
-                <?php if ($language = $value->lang()): ?>
-                <span class="language"><?php echo $language; ?></span>
+            <?php if (($filterLocale == 0) || ($valueLang == '') || ($valueLang == $lang)): ?>
```

Comment:
I'd have expected the actual filtering of values to happen at the resource representation's `displayValues` method. Partially just because we already have _other_ "lang-filtering" functionality there in `value()`.

If it's going to be here (or anywhere, I guess), then we have to handle some things that the `value()` implementation already does, like comparing langs case-insensitively.

Reply:
You're right we should be able to filter this when retrieving the values. I don't see immediately if `AbstractResourceEntityRepresentation::displayValues()` uses exactly the same implementation as `AbstractResourceEntityRepresentation::value()` for each value, but I'll have a closer look.

Reply:
ed9336a introduces this filtering being done within `AbstractResourceEntityRepresentation` display methods.